packages/kobotoolbox/test/util.test.js

```diff
@@ -25,100 +26,64 @@ const defaultObjects = [
   { uid: '6' },
 ];
 
-const responseWithPagination = (
-  url,
-  pageSize = 1000,
-  start = 0,
-  objects = defaultObjects
-) => {
-  const s = parseInt(start);
-  const l = parseInt(pageSize);
-  const results = objects.slice(s, s + l);
-  const count = objects.length;
-  const previous = `${url}?format=json&start=${s}&limit=${l}`;
-  let next = null;
-  if (count > s + l) {
-    next = `${url}?format=json&start=${s + l}&limit=${l}`;
-
-    return {
-      next,
-      count,
-      previous,
-      results,
-    };
-  } else {
-    return { results, count, previous, next };
-  }
-};
-
 describe('responseWithPagination', () => {
   const items = [1, 2, 3];
   it('should return the first item', () => {
     const start = 0;
-    const pageSize = 1;
-    const { next, results } = responseWithPagination(
-      'www',
-      pageSize,
+    const limit = 1;
+    const { results } = responseWithPagination('www', items, {
+      limit,
       start,
-      items
-    );
+    });
     expect(results).to.eql([1]);
   });
-  it('should return 2 items with pageSize 2', () => {
+  it('should return 2 items with limit 2', () => {
     const start = 0;
-    const pageSize = 2;
-    const { next, results } = responseWithPagination(
-      'www',
-      pageSize,
+    const limit = 2;
+    const { results } = responseWithPagination('www', items, {
+      limit,
       start,
-      items
-    );
+    });
     expect(results).to.eql([1, 2]);
   });
   it('should return 2 items with start 1 and pageSize 2', () => {
     const start = 1;
     const pageSize = 2;
-    const { next, results } = responseWithPagination(
-      'www',
+    const { results } = responseWithPagination('www', items, {
       pageSize,
       start,
-      items
-    );
+    });
     expect(results).to.eql([2, 3]);
   });
   it('should return all items', () => {
     const start = 0;
     const pageSize = 3;
-    const { next, results } = responseWithPagination(
-      'www',
+    const { next, results } = responseWithPagination('www', items, {
       pageSize,
       start,
-      items
-    );
+    });
+
     expect(results).to.eql([1, 2, 3]);
   });
   it('should return all items if pageSize is greater', () => {
     const start = 0;
     const pageSize = 100;
-    const { next, results } = responseWithPagination(
-      'www',
+    const { next, results } = responseWithPagination('www', items, {
       pageSize,
       start,
-      items
-    );
+    });
     expect(results).to.eql([1, 2, 3]);
   });
   it('should return the correct next link', () => {
```

Comment:
hmm, I think you should assert the `next` link in every test. Even if undefined. Then you need to rethink the purpose of this test.

because it's hard to test the url string, you might prefer to return like a `_next` object with all the key/value pairs. That's not used by the adaptor, but it can be used in these tests to ensure we're getting the correct behaviour